app/src/androidTest/java/com/github/se/cyrcle/model/user/UserViewModelTest.kt

```diff
@@ -0,0 +1,133 @@
+package com.github.se.cyrcle.model.user
+
+import UserViewModel
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import com.google.firebase.FirebaseApp
+import com.google.firebase.firestore.FirebaseFirestore
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+
+class UserViewModelTest {
+
+  private lateinit var userViewModel: UserViewModel
+  private lateinit var userRepositoryFirestore: UserRepositoryFirestore
+
+  @Before
+  fun setUp() = runTest {
+    // Initialize Firebase and Firestore
+    val context = ApplicationProvider.getApplicationContext<Context>()
+    FirebaseApp.initializeApp(context)
+
+    val firestore = FirebaseFirestore.getInstance()
+
+    // Initialize the UserRepositoryFirestore with Firestore
+    userRepositoryFirestore = UserRepositoryFirestore(firestore)
+
+    // Initialize the UserViewModel with the real repository
+    userViewModel = UserViewModel(userRepositoryFirestore)
+    userViewModel.addUser(user1)
+  }
+
+  @Test
+  fun fetchUserById_updates_selectedUser_on_success() = runTest {
+    val countDownLatch = CountDownLatch(1)
+
+    // Act: Fetch a user by ID (make sure the user exists in Firestore)
+    userViewModel.fetchUserById("user1")
+
+    // Wait for the asynchronous call to complete (up to 5 seconds)
+    countDownLatch.await(5, TimeUnit.SECONDS)
+
+    // Assert: Check if the user is updated correctly in the StateFlow
+    val fetchedUser = userViewModel.user.first()
+    assertNotNull(fetchedUser)
+    assertEquals("user1", fetchedUser?.userId)
+    assertEquals("john_doe", fetchedUser?.username)
+  }
+
+  val newUser =
```

Comment:
Those user declaration can be moved into the above mentioned `TestInstancesUser` object